src/route-loader.js

```diff
@@ -46,12 +62,22 @@ export class TemplatingRouteLoader extends RouteLoader {
       return childContainer.get(Router);
     };
 
+    if (isPromise) {
+      return config.moduleId.then(viewModelClass => {
```

Comment:
So if you do `{ moduleId: require('my-route') }` wouldn't this code block eagerly load the route at configuration?

Reply:
I think all bundlers will turn `require('module-id')` into a synchronous call, and what it resolves into (`string`, `Function`, `Promise` ) will be handled just fine.

Reply:
So what's the promise use case example then?

Reply:
Based on a spec of dynamic import module, bundlers use `import(string)` as a sign for code split point. This is to enable that scenario for big modules

Reply:
In that case, wouldn't this eagerly load instead of lazily?

Reply:
From the `loadRoute` code in `TemplatingRouteLoader` (also I tested, but 'd love to have you confirm this), I understand that the promise will only be resolved if we navigate to that specific route. So it's still lazily loaded I think.

Reply:
Just passing `import('module')` won't tell the bundler to split. So just passing a promise is the same as passing the class directly (promises aren't lazy). The correct use would be to pass a function that returns a promise the the module: `() => import('module')`

Reply:
@obedm503 You are correct. @davismj Sorry I overlooked this piece of code. I kept thinking that I did `() => import('module')...` instead of `import('...')` . Thanks @obedm503 for the correction

Reply:
Also makes me realize I didn't handle this case well: currently cannot differentiate between a route view model class and a function that returns a promise resolves to a class, will fix this.

Reply:
Sounds like this will need some love.

Reply:
we can ... enforce it the other way around, always invoke the function, if returns a promise, we know what to do ... 😄 in case it's too hard